OPM/opm-rule-onboarding.ps1

```diff
@@ -0,0 +1,540 @@
+
+
+#region [Logon]
+	# Get Credentials to Login
+	# ------------------------
+	$caption = "OPM Rule Onboarding Utility"
+	$msg = "Enter your User name and Password"; 
+	$creds = $Host.UI.PromptForCredential($caption,$msg,"","")
+	if ($creds -ne $null)
+	{
+		if($AuthType -eq "radius" -and ![string]::IsNullOrEmpty($OTP))
+		{
+			$g_LogonHeader = $(Get-LogonHeader -Credentials $creds -RadiusOTP $OTP)
+		}
+		else
+		{
+			$g_LogonHeader = $(Get-LogonHeader -Credentials $creds)
+		}
+	}
+	else { 
+		Log-Msg -Type Error -MSG "No credentials were entered." -Footer
+		exit
+	}
+#endregion
+
+#region [Read Accounts CSV file and Create Accounts]
+	If([string]::IsNullOrEmpty($CsvPath))
+	{
+		$CsvPath = OpenFile-Dialog($g_CsvDefaultPath)
+	}
+	$delimiter = $(If ($CsvDelimiter -eq "Comma") { "," } else { "`t" } )
+	$rulesCSV = Import-CSV $csvPath -Delimiter $delimiter
+	$rowCount = $($rulesCSV.Safe.Count)
+	$counter = 0
+	$successCounter = 0
+	Log-Msg -Type Info -MSG "Starting to Onboard $rowCount rules." -SubHeader
+	ForEach ($rule in $rulesCSV)
+	{
+		if ($null -ne $rule)
+		{
+			try {
+				# Check mandatory fields
+				If([string]::IsNullOrEmpty($rule.PlatformId)) { throw "Missing mandatory field: Platform ID" }
+				If([string]::IsNullOrEmpty($rule.Command)) { throw "Missing mandatory field: Command" }
+				If([string]::IsNullOrEmpty($rule.UserName)) { throw "Missing mandatory field: UserName" }
+				
+				$rule.PlatformId = (Get-TrimmedString $rule.PlatformID)
+				$objRule = "" | Select-Object "Command", "CommandGroup", "PermissionType", "Restrictions", "UserName"
+				$objRule.Command = (Get-TrimmedString $rule.Command)
+				$objRule.CommandGroup = (Get-TrimmedString $rule.CommandGroup)
+				$objRule.PermissionType = (Get-TrimmedString $rule.PermissionType)
+				$objRule.Restrictions = (Get-TrimmedString $rule.Restrictions)
+				$objRule.UserName = (Get-TrimmedString $rule.UserName)
+				
+				if ([string]::IsNullOrEmpty($objRule.CommandGroup)) { 
+					$objRule.CommandGroup = $false
+				} elseif ($objRule.CommandGroup.ToLower() -eq "yes" -or $objRule.CommandGroup.ToLower() -eq "true") {
+					$objRule.CommandGroup = $true
+				} else {
+					$objRule.CommandGroup = $false
+				}
+				
+				if ([string]::IsNullOrEmpty($objRule.PermissionType)) { 
+					$objRule.PermissionType = "Allow"
+				} elseif ($objRule.permissionType.ToLower() -eq "allow" -or $objRule.permissionType.ToLower() -eq "deny") {
+					$objRule.PermissionType = (Get-Culture).TextInfo.ToTitleCase($objRule.PermissionType)
+				} else {
+					throw "Invalid field: PermissionType"
+				}
+				
+				if ($objRule.commandGroup -eq $false -and -Not $objRule.command.StartsWith("/") -and -Not $objRule.command -eq ".*") {
+					Write-Warning "Rule Number $counter : It is a recommended security best practice to define commands with an absolute path. Please reconfigure rules like (cat /etc/passwd) to (/usr/bin/cat /etc/passwd)" -WarningAction Inquire
```

Comment:
You can use here the Log-Msg as well with a type of Warning
This prevents script for pausing and keeps the information in the log as well as on screen